### ISSUE TITLE
Shipping method IDs changed between WC 2.6 beta 1 and beta 2 ?

### ISSUE THREAD

**Issue by DavidAnderson684** [woothemes/woocommerce#10918]
Hi,

Did the format of the shipping method values used on the checkout page, and POSTed via in the shipping_method field, just change between WC 2.6 beta 1 and beta 2? It's not mentioned on the blog post - https://woocommerce.wordpress.com/2016/05/13/woocommerce-2-6-beta-2/ - and I can't spot the changeset on a quick look through - https://github.com/woothemes/woocommerce/compare/2.6.0-beta-1...master - but it looks like the format, in beta 1, was Method_idInstance_id, but now there's a colon: Method_id:Instance_id

Was this intentional? It'd be a good thing to mention. If it's not intentional, I'll hold off updating my code again...

**Comment by mikejolley** [comment]
> Added functions to give shipping method instances consistent rate IDs

This was mentioned. 

https://github.com/woothemes/woocommerce/commit/2b1442c3f0b436495e62c40a98c21fcbdfe477ba

**Comment by mikejolley** [comment]
Also makes it consistent with rate IDs from Table Rate Shipping (current version).

p.s. no need to post in multiple places. I deleted your blog comment.

**Comment by DavidAnderson684** [comment]
Hi Mike,

Sorry for the double post - I posted it here after asking it on the blog, because after commenting on the beta 2 blog post, I then remembered that my comment on something else on the beta 1 blog post did not see a reply after a few days, so after posting on the blog, I then thought "this is the wrong place - they want this stuff in Github, not as blog comments".

"This was mentioned." - By "mentioned", I meant the blog post announcing beta 2. I saw "Added get_rate_id() method for consistent rate nams" on the list of commits, of course - but I don't think I'm being too weird if I say that a commit message indicating a new getter method didn't light up the "potential breaking data format change" warning lights in my head. I think extension devs would be really grateful for something like "[Breaking format change]" in the commit message, as a big saver of man hours (and I'm sure we'd understand that sometimes they'd slip through without that - but every little helps).
